.githooks/pre-push

```diff
@@ -12,15 +12,25 @@ if lines.empty?
 end
 
 puts "1. Lint RBI files"
-system("bundle exec rubocop #{files.join(" ")}") ? continue : exit(1)
+unless system("bundle exec rubocop #{files.join(" ")}")
+  exit(1)
+end
 puts "2. Typecheck RBI files"
-system("scripts/check_types #{files.join(" ")}") ? continue : exit(1)
+unless system("scripts/check_types #{files.join(" ")}")
+  exit(1)
+end
 puts "3. Check new RBI files belong to public gems"
-system("scripts/check_gems_are_public #{files.join(" ")}") ? continue : exit(1)
+unless system("scripts/check_gems_are_public #{files.join(" ")}")
+  exit(1)
+end
 puts "4. Lint index.json"
-system("scripts/lint_index_json") ? continue : exit(1)
+unless system("scripts/lint_index_json") 
+  exit(1)
+end
 puts "5. Check index.json entries match rbi/annotations/*.rbi files"
-system("scripts/match_index_entries_with_rbis") ? continue : exit(1)
+unless system("scripts/match_index_entries_with_rbis")
+  exit(1)
+end
```

Comment:
Can we run all the tests rather than exiting early. So if there is a problem with a a RBI _and_ a lint error in the `index.json` file I won't need to push twice to see it?

Reply:
Okay! Would the best way to approach this to be some sort of variable like `success = true` and set it to false if any of the tests fail in place of `exit(1)` so for example:
```
unless system("scripts/match_index_entries_with_rbis")
  success = false
end
```
and then at the end having logic to `exit(0)` if success and `exit(1)` otherwise?